### PR TITLE
Break FD hacks

### PR DESCRIPTION
**Pull request by ngc92**
use pipes instead of unlinked files to guarantee unidirectional transfers

### CODE REVIEW

File: python/pygpubench/__init__.py [hunk 1]
```diff
@@ -49,15 +49,15 @@ def do_bench_impl(out_file: str, qualname: str, test_generator: TestGeneratorInt
     try:
         with DeterministicContext():
             _pygpubench.do_bench(
-                out_file,
+                out_fd.fileno(),
+                signature.fileno(),
                 qualname,
```
[the remark this answers is not in the document]
Comment (ngc92) [reply python/pygpubench/__init__.py]: conn.detach() is hallucinated.
